masakhwe/RepairHub
# #165575975 Detailed repair request views

### What does this PR do?
Enables a user to view a list of saved repair requests and when he clicks on any of the requests, he is able to view its details

### What are the tasks to be completed?

- Ensure repairs list and repair detail pages load successfully

- Ensure correct templates are used for both list and detail views

- Ensure the saved repair requests can be retrieved from the database and displayed on list page

- Ensure a user can view the details of a single request when he clicks on it on the list page

### How Should this be manually tested?
Create a virtual environment
`virtualenv env`

Clone the repo
`git clone https://github.com/masakhwe/RepairHub.git`

Checkout to ft-repair-request-detailed-view-165575975 branch
`git checkout ft-repair-request-detailed-view-165575975`

Install the dependancies
`pip install -r requirements.txt`

Run pytest and ensure all tests are passing
`pytest -v`

Start the development server by running the command below
`python manage.py runserver`

Open your browser and enter the following to view the current repair requests list
`http://localhost:8000/repairs`

Click on any of the requests to view its details

### Screenshots
![failing_tests_detailed_repair_request](https://user-images.githubusercontent.com/24722920/62841624-36702d80-bcb3-11e9-8606-aea0fbb33dbf.png)

![passing_tests_detailed_repair_request](https://user-images.githubusercontent.com/24722920/62841631-47b93a00-bcb3-11e9-8b7a-43bda68d11f8.png)

![repairs_list](https://user-images.githubusercontent.com/24722920/62841643-569fec80-bcb3-11e9-9b82-bec3d0e535d7.png)

![repair_detail](https://user-images.githubusercontent.com/24722920/62841647-628bae80-bcb3-11e9-94fa-79897d219cce.png)

### What are the relevant pivotal tracker stories
[#165575975](https://www.pivotaltracker.com/story/show/165575975)
